contents/stacks_and_queues/code/rust/Queue.rs

```diff
@@ -0,0 +1,47 @@
+use std::collections::VecDeque;
+
+struct Queue<T> {
+    list: VecDeque<T>
+}
+
+impl<T> Queue<T> {
+    fn new() -> Self {
+       Queue{
+           list: VecDeque::new(),
+       }
+    }
+
+    // Note that this returns a reference to the value
+    // This is in contrast to dequeue which gives ownership of the value
+    fn front(&self) -> Option<&T> {
+        if self.list.len() > 0 {
+            Some(&self.list[0])
```

Comment:
It might best to use the methods in VecDeque for this
```suggestion
            self.list.front()
```